### PR TITLE
#167 - Created wrap for Headers

### PR DESCRIPTION
**Pull request by olenagerasimova**
For #167 created abstract `Headers` implementation.

### CODE REVIEW

**Comment by 0crat** [comment]
This pull request [#168](https://github.com/artipie/http/issues/168) is assigned to @paulodamaso[/z](https://www.0crat.com/u/paulodamaso), here is [why](https://www.0crat.com/footprint/CT2E6TK9B/13732322-0b36-403d-ab56-7b2761fba36b); the budget is 15 minutes, see [§4](http://www.zerocracy.com/policy.html#4); please, read [§27](http://www.zerocracy.com/policy.html#27) and when you decide to accept the changes, inform @g4s8[/z](https://www.0crat.com/u/g4s8) (the architect) right in this ticket; if you decide that this PR should not be accepted ever, also inform the architect; this [blog post](http://www.yegor256.com/2015/02/09/serious-code-reviewer.html) will help you understand what is expected from a code reviewer; there will be a monetary reward for this job

**Comment by 0crat** [comment]
@g4s8[/z](https://www.0crat.com/u/g4s8) I see that you unassigned this issue; the order is still assigned to @paulodamaso[/z](https://www.0crat.com/u/paulodamaso) though; to cancel the order use `refuse`, as in [§6](http://www.zerocracy.com/policy.html#6)

**Comment by 0crat** [comment]
This pull request [#168](https://github.com/artipie/http/issues/168) is assigned to @olegmoz[/z](https://www.0crat.com/u/olegmoz), here is [why](https://www.0crat.com/footprint/CT2E6TK9B/9ddd3bc8-f784-469b-988d-ec266cf0151c); the budget is 15 minutes, see [§4](http://www.zerocracy.com/policy.html#4); please, read [§27](http://www.zerocracy.com/policy.html#27) and when you decide to accept the changes, inform @g4s8[/z](https://www.0crat.com/u/g4s8) (the architect) right in this ticket; if you decide that this PR should not be accepted ever, also inform the architect; this [blog post](http://www.yegor256.com/2015/02/09/serious-code-reviewer.html) will help you understand what is expected from a code reviewer; @olegmoz[/z](https://www.0crat.com/u/olegmoz) is not a member of this project yet, but they can request to join, as [§1](http://www.zerocracy.com/policy.html#1) explains; there will be no monetary reward for this job

File: src/main/java/com/artipie/http/Headers.java [hunk 1]
```diff
@@ -100,4 +100,47 @@ public void forEach(final Consumer<? super Map.Entry<String, String>> action) {
...
+
+        @Override
+        public void forEach(final Consumer<? super Map.Entry<String, String>> action) {
+            this.origin.forEach(action);
+        }
+
+        @Override
+        public Spliterator<Map.Entry<String, String>> spliterator() {
+            return this.origin.spliterator();
+        }
```
Comment (olegmoz) [inline src/main/java/com/artipie/http/Headers.java:144]: @olenagerasimova please change all overridden methods to be final

**Comment by olenagerasimova** [comment]
@olegmoz thanks, corrected

**Comment by olenagerasimova** [comment]
@g4s8 could you merge this please?

**Comment by g4s8** [comment]
@olenagerasimova the build is failing, please check https://github.com/artipie/http/pull/168/checks?check_run_id=733991458

**Comment by olenagerasimova** [comment]
@g4s8 corrected, please merge

**Comment by 0crat** [comment]
@sereshqua[/z](https://www.0crat.com/u/sereshqua) please review this job completed by @olegmoz[/z](https://www.0crat.com/u/olegmoz), as in [§30](http://www.zerocracy.com/policy.html#30); the job will be fully closed and all payments will be made when the quality review is completed

**Comment by sereshqua** [comment]
@olegmoz please make you will find at least 3 issues during next CR, thanks!

**Comment by sereshqua** [comment]
@0crat quality acceptable

**Comment by 0crat** [comment]
> @0crat quality acceptable  ([here](https://github.com/artipie/http/issues/168#issuecomment-639993719))

@sereshqua There is an unrecoverable failure on my side. Please, submit it [here](https://github.com/zerocracy/farm/issues):

```
PID: 4@d37b4fb8-9d4e-4af1-bac4-234738e082f0, thread: PQ-CT2E6TK9B
com.zerocracy.farm.reactive.StkRuntime[76] java.lang.IllegalStateException: org.codehaus.groovy.runtime.InvokerInvocationException in com.zerocracy.stk.pm.cost.pay_cash
org.codehaus.groovy.reflection.CachedMethod[105] org.codehaus.groovy.runtime.InvokerInvocationException: java.io.IOException: Failed to add transaction
com.zerocracy.pm.cost.Ledger[133] java.io.IOException: Failed to add transaction
org.postgresql.jdbc.PgConnection[767] org.postgresql.util.PSQLException: This connection has been closed.
```

[0.53.15](https://github.com/zerocracy/farm): CID: [551a0402-7140-4cf5-930c-238c59801b9e](https://www.0crat.com/footprint/CT2E6TK9B/551a0402-7140-4cf5-930c-238c59801b9e), Type: "Pay cash"

**Pull request merged**